src/tracer.ts

```diff
@@ -224,12 +224,16 @@ export const trace =
       return performPromisifyType(err, data, type, callback);
     }
     context.__wrappedByLumigo = true;
+    let pStartTrace = Promise.resolve();
+    let functionSpan;
+    if (!isSwitchedOff()) {
```

Comment:
I think that this `if` should be the first line in this function (`trace`), and it should immediately stop everything (all the code lines in here should not run!)